### ISSUE TITLE
Unable to create stoploss order because of insufficient funds, on gate.io, when using leverage

### ISSUE THREAD

**Issue by gaardiolor** [freqtrade/freqtrade#8341]
## Describe your environment

  * Operating system: Docker
  * Python Version: 3.10
  * CCXT version: 1.92.84
  * Freqtrade Version: docker-2023.3.dev-764d5507
  
## Describe the problem:

In a running freqtrade, but (also after updating and restarting), errors are being spammed in the log. The errors indicate that:
- Freqtrade is trying to create a stop order for an already open position
- This fails because the leverage can't be set to 2 (not sure why, it's a stop order) because of insufficient funds (not sure why, it's a stop order). I have $0.97 available in my futures account.

### Steps to reproduce:
I'm not sure about the exact circumstances, I think freqtrade should be able to create the stop order. I'm able to create a stop order manually on gate.io, (although it looks like it's a different type of stop order according to the UI, compared to the stop orders freqtrade creates - not sure why..) I don't know why freqtrade is trying to set the leverage when creating a stop order, or why the exchange is responding with insufficient funds.
  
### Observed Results:
See above. I expect stop orders to be created.. I do not think I need funds to create stop orders. 

### Relevant code exceptions or logs

  ```
2023-03-17 20:03:39,366 - freqtrade.worker - INFO - Changing state to: RUNNING
2023-03-17 20:03:39,366 - freqtrade.rpc.rpc_manager - INFO - Sending rpc message: {'type': startup, 'status': "*Exchange:* `gateio`\n*Stake per trade:* `unlimited USDT`\n*Minimum ROI:* `{'0': 100}`\n*Stoploss:* `-0.08`\n*Position adjustment:* `Off`\n*Timeframe:* `5m`\n*Strategy:* `Strat`"}
2023-03-17 20:03:39,409 - freqtrade.rpc.rpc_manager - INFO - Sending rpc message: {'type': startup, 'status': "Searching for USDT pairs to buy and sell based on [{'StaticPairList': 'StaticPairList'}, {'AgeFilter': 'AgeFilter - Filtering pairs with age less than 10 days'}, {'PrecisionFilter': 'PrecisionFilter - Filtering untradable pairs.'}, {'PriceFilter': 'PriceFilter - Filtering pairs priced below 1.0%.'}]"}
2023-03-17 20:03:39,519 - freqtrade.persistence.trade_model - INFO - Found open trade: Trade(id=1464, pair=BTC/USDT:USDT, amount=0.01610000, is_short=False, leverage=2.0, open_rate=24232.80000000, open_since=2023-03-16 00:50:09)
2023-03-17 20:03:39,519 - freqtrade.persistence.trade_model - INFO - Found open trade: Trade(id=1470, pair=NEAR/USDT:USDT, amount=197.00000000, is_short=True, leverage=2.0, open_rate=1.95370000, open_since=2023-03-16 16:50:12)
2023-03-17 20:03:39,519 - freqtrade.persistence.trade_model - INFO - Found open trade: Trade(id=1471, pair=ADA/USDT:USDT, amount=1170.00000000, is_short=True, leverage=2.0, open_rate=0.32752000, open_since=2023-03-16 19:25:08)
2023-03-17 20:03:39,519 - freqtrade.persistence.trade_model - INFO - Found open trade: Trade(id=1472, pair=ETH/USDT:USDT, amount=0.22000000, is_short=False, leverage=2.0, open_rate=1683.40000000, open_since=2023-03-16 19:30:10)
2023-03-17 20:03:39,519 - freqtrade.persistence.trade_model - INFO - Found open trade: Trade(id=1473, pair=DOGE/USDT:USDT, amount=5350.00000000, is_short=True, leverage=2.0, open_rate=0.07082200, open_since=2023-03-16 19:35:09)
2023-03-17 20:03:39,521 - freqtrade.freqtradebot - INFO - Updating 2 open orders.
2023-03-17 20:03:39,761 - freqtrade.freqtradebot - INFO - Found open order for Trade(id=1464, pair=BTC/USDT:USDT, amount=0.01610000, is_short=False, leverage=2.0, open_rate=24232.80000000, open_since=2023-03-16 00:50:09)
2023-03-17 20:03:40,074 - freqtrade.freqtradebot - INFO - Found open order for Trade(id=1472, pair=ETH/USDT:USDT, amount=0.22000000, is_short=False, leverage=2.0, open_rate=1683.40000000, open_since=2023-03-16 19:30:10)
2023-03-17 20:03:43,134 - freqtrade.exchange.common - WARNING - _set_leverage() returned exception: "Could not set leverage due to InsufficientFunds. Message: gateio {"label":"INSUFFICIENT_AVAILABLE"}". Retrying still for 4 times.
2023-03-17 20:03:43,377 - freqtrade.exchange.common - WARNING - _set_leverage() returned exception: "Could not set leverage due to InsufficientFunds. Message: gateio {"label":"INSUFFICIENT_AVAILABLE"}". Retrying still for 3 times.
2023-03-17 20:03:43,629 - freqtrade.exchange.common - WARNING - _set_leverage() returned exception: "Could not set leverage due to InsufficientFunds. Message: gateio {"label":"INSUFFICIENT_AVAILABLE"}". Retrying still for 2 times.
2023-03-17 20:03:43,871 - freqtrade.exchange.common - WARNING - _set_leverage() returned exception: "Could not set leverage due to InsufficientFunds. Message: gateio {"label":"INSUFFICIENT_AVAILABLE"}". Retrying still for 1 times.
2023-03-17 20:03:44,114 - freqtrade.exchange.common - WARNING - _set_leverage() returned exception: "Could not set leverage due to InsufficientFunds. Message: gateio {"label":"INSUFFICIENT_AVAILABLE"}". Giving up.
2023-03-17 20:03:44,115 - freqtrade.exchange.common - WARNING - stoploss() returned exception: "Could not set leverage due to InsufficientFunds. Message: gateio {"label":"INSUFFICIENT_AVAILABLE"}". Giving up.
2023-03-17 20:03:44,115 - freqtrade.freqtradebot - ERROR - Unable to place a stoploss order on exchange.
Traceback (most recent call last):
  File "/home/ftuser/.local/lib/python3.10/site-packages/ccxt/base/exchange.py", line 620, in fetch
    response.raise_for_status()
  File "/home/ftuser/.local/lib/python3.10/site-packages/requests/models.py", line 1021, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: https://api.gateio.ws/api/v4/futures/usdt/positions/NEAR_USDT/leverage?leverage=2.0

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/freqtrade/freqtrade/exchange/exchange.py", line 2418, in _set_leverage
    res = self._api.set_leverage(symbol=pair, leverage=leverage)
  File "/home/ftuser/.local/lib/python3.10/site-packages/ccxt/gate.py", line 3746, in set_leverage
    response = getattr(self, method)(self.extend(request, query))
  File "/home/ftuser/.local/lib/python3.10/site-packages/ccxt/base/exchange.py", line 500, in inner
    return entry(_self, **inner_kwargs)
  File "/home/ftuser/.local/lib/python3.10/site-packages/ccxt/base/exchange.py", line 2610, in request
    return self.fetch2(path, api, method, params, headers, body, config, context)
  File "/home/ftuser/.local/lib/python3.10/site-packages/ccxt/base/exchange.py", line 2607, in fetch2
    return self.fetch(request['url'], request['method'], request['headers'], request['body'])
  File "/home/ftuser/.local/lib/python3.10/site-packages/ccxt/base/exchange.py", line 636, in fetch
    skip_further_error_handling = self.handle_errors(http_status_code, http_status_text, url, method, headers, http_response, json_response, request_headers, request_body)
  File "/home/ftuser/.local/lib/python3.10/site-packages/ccxt/gate.py", line 4404, in handle_errors
    self.throw_exactly_matched_exception(self.exceptions['exact'], label, feedback)
  File "/home/ftuser/.local/lib/python3.10/site-packages/ccxt/base/exchange.py", line 2853, in throw_exactly_matched_exception
    raise exact[string](message)
ccxt.base.errors.InsufficientFunds: gateio {"label":"INSUFFICIENT_AVAILABLE"}

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/freqtrade/freqtrade/exchange/common.py", line 165, in wrapper
    return f(*args, **kwargs)
  File "/freqtrade/freqtrade/exchange/exchange.py", line 2423, in _set_leverage
    raise TemporaryError(
freqtrade.exceptions.TemporaryError: Could not set leverage due to InsufficientFunds. Message: gateio {"label":"INSUFFICIENT_AVAILABLE"}

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/ftuser/.local/lib/python3.10/site-packages/ccxt/base/exchange.py", line 620, in fetch
    response.raise_for_status()
  File "/home/ftuser/.local/lib/python3.10/site-packages/requests/models.py", line 1021, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: https://api.gateio.ws/api/v4/futures/usdt/positions/NEAR_USDT/leverage?leverage=2.0
  ```

Trade json:
```
[
    {
        "trade_id": 1470,
        "pair": "NEAR/USDT:USDT",
        "base_currency": "NEAR",
        "quote_currency": "USDT",
        "is_open": true,
        "exchange": "gateio",
        "amount": 197.0,
        "amount_requested": 197.95906694,
        "stake_amount": 192.43945,
        "max_stake_amount": 192.43945,
        "strategy": "Strat",
        "enter_tag": "buy_signal_short",
        "timeframe": 5,
        "fee_open": 0.0005,
        "fee_open_cost": 0.19243944999999998,
        "fee_open_currency": "USDT",
        "fee_close": 0.0005,
        "fee_close_cost": null,
        "fee_close_currency": null,
        "open_date": "2023-03-16 16:50:12",
        "open_timestamp": 1678985412717,
        "open_rate": 1.9537,
        "open_rate_requested": 1.9513,
        "open_trade_value": 384.68646055,
        "close_date": null,
        "close_timestamp": null,
        "realized_profit": 0.0,
        "realized_profit_ratio": null,
        "close_rate": null,
        "close_rate_requested": null,
        "close_profit": null,
        "close_profit_pct": null,
        "close_profit_abs": null,
        "trade_duration_s": null,
        "trade_duration": null,
        "profit_ratio": null,
        "profit_pct": null,
        "profit_abs": null,
        "exit_reason": null,
        "exit_order_status": null,
        "stop_loss_abs": 1.9966,
        "stop_loss_ratio": -0.0013030621961607203,
        "stop_loss_pct": -0.13030621961607203,
        "stoploss_order_id": null,
        "stoploss_last_update": "2023-03-17 06:18:38",
        "stoploss_last_update_timestamp": 1679033918282,
        "initial_stop_loss_abs": 2.0319,
        "initial_stop_loss_ratio": -0.08,
        "initial_stop_loss_pct": -8.0,
        "min_rate": 1.9272,
        "max_rate": 2.0939,
        "leverage": 2.0,
        "interest_rate": 0.0,
        "liquidation_price": 2.813735,
        "is_short": true,
        "trading_mode": "futures",
        "funding_fees": 0.11813499,
        "open_order_id": null,
        "orders": [
            {
                "amount": 197.0,
                "safe_price": 1.9537,
                "ft_order_side": "sell",
                "order_filled_timestamp": 1678985413538,
                "ft_is_entry": true,
                "pair": "NEAR/USDT:USDT",
                "order_id": "271929193324",
                "status": "closed",
                "average": 1.9537,
                "cost": 384.8789,
                "filled": 197.0,
                "is_open": false,
                "order_date": "2023-03-16 16:50:12",
                "order_timestamp": 1678985412590,
                "order_filled_date": "2023-03-16 16:50:13",
                "order_type": "limit",
                "price": 1.9513,
                "remaining": 0.0
            }
        ]
    }
]

```

**Comment by xmatthias** [comment]
ccxt version 1.92.84 seems wrong - that's a version that's probably 100ds of versions behind the version corresponding to the freqtrade version.
Are you overriding the dockerfile - and downgrading ccxt intentionally? (that's the only way you can get to such a version in a docker context).


That said - I've just tested this (with an uptodate ccxt version) - and creating a stoploss order works perfectly fine (tested multiple times, including going "all in" on one trade to exclude potential problems caused by having insufficient collateral available, as well as in a trailing scenario, where the stop was pulled up several times.

To me, it seems to work just fine (on an uptodate ccxt version, though).

**Comment by gaardiolor** [comment]
This is strange.. I did not overwrite the dockerfile. Maybe an image layer got stuck in my docker setup.. weird. The build section is commented out in my docker-compose.yml

I did a `docker-compose down --rmi all -v --remove-orphans` and pulled again. Now I'm on docker-2023.3.dev-186fe593 and CCXT 2.9.12

Let's assume it's fixed, if not I'll open a new issue. Thanks!

**Comment by gaardiolor** [comment]
Unfortunately, the issue came back.. guessed it's better to re-open this issue as it looks exactly the same, hope that's ok.

docker-2023.3.dev-186fe593
CCXT 2.9.12

Before starting this new version of the container (including ccxt) I closed all trades directly on gate.io. After starting I `/delete`'d all the trades via telegram. 

What's interesting:
- The first stop orders work.. they seem to stop working after position 5 has been opened (with `"max_open_trades": 5`) . 
- I have `"tradable_balance_ratio": 0.99` , yet my balance when I have this problem was less than 0.1%
- I'm still not sure why freqtrade / CCXT is trying to adjust leverage when it's a stop order..

**Comment by xmatthias** [comment]
> I'm still not sure why freqtrade / CCXT is trying to adjust leverage when it's a stop order..

i don't see why this should be problematic. 
we're not actually changing leverage, but leave it as it was for this pair/trade.

The only case where leverage would actually change is if you manually messed with leverage settings for that pair on the exchange - which is however unsupported (freqtrade assumes a futures account is exclusively used by it. this is also clearly documented in the [leverage](https://www.freqtrade.io/en/latest/leverage/) section (first Note box).

This is also my best assumption as to why you're encountering this problem - as i went "all in" on my account - without having this problem once.

**Comment by gaardiolor** [comment]
I did not change leverage settings on the exchange.. I've got a `leverage` definition in the strategy that always returns 2. 

What I'm seeing is that, after position 5 has been opened, I have 1% funds in my account - corresponding to `tradable_balance_ratio`. One hour later, without any new trades, this dropped to 0.1% and I'm getting the errors. 

Some process is eating the remaining 1%. This is the only bot on the account..

Have to dive into this a bit deeper, and try to link a message in the log to the exact time the balance dropped.

**Comment by xmatthias** [comment]
Your best bet will be the exchange website - looking for any orders or whatnot that might be consuming your funds.

Freqtrade itself won't consume them unless you also see something about orders being created in the logs.
From the behavior - it would sound like cross margin behavior - where losing positions consume "more" of the available stake, to eventually cover the position.

We do use explicitly isolated margin though - and I'm not aware of any extra settings necessary for gate.

**Comment by gaardiolor** [comment]
I've also already had configured margin mode to isolated just to be sure, also not getting any liquidations, so that should not be it.. But I'll do a few tests. 

- Topup available balance a bit
- Stop freqtrade, run fetch_balance() every few seconds, log output, see if available balance changes, if not
- Start freqtrade, try to link messages in the logfile to a drop in available balance

I'll test this and report back.

**Comment by xmatthias** [comment]
I've now added an exception-handler for the raised exception - as it's not relevant if it fails when called from stops - but calling it will still keep the above mentioned safety of "hasn't been messed with".

**Comment by gaardiolor** [comment]
Ok, so I've witnessed my balance change. 

At first, `free` increased and `used` dropped. 

```
2023-03-21_22.20.08 balance USDT:      {"free": 6.671774538136, "used": 867.7562582875, "total": 874.428032825636}
2023-03-21_22.20.14 balance USDT:      {"free": 7.029237538136, "used": 867.3987952875, "total": 874.428032825636}
```

Corresponding freqtrade log (timestams are 1 hour behind):

```
2023-03-21 21:20:07,697 - freqtrade.freqtradebot - INFO - Cancelling current stoploss on exchange for pair SOL/USDT:USDT (orderid:39152684) in order to add another one ...
2023-03-21 21:20:09,230 - freqtrade.freqtradebot - INFO - Cancelling current stoploss on exchange for pair BTC/USDT:USDT (orderid:39152686) in order to add another one ...
2023-03-21 21:20:10,787 - freqtrade.freqtradebot - INFO - Cancelling current stoploss on exchange for pair FTM/USDT:USDT (orderid:39152688) in order to add another one ...
2023-03-21 21:20:12,302 - freqtrade.freqtradebot - INFO - Cancelling current stoploss on exchange for pair DOT/USDT:USDT (orderid:39152691) in order to add another one ...
2023-03-21 21:20:13,839 - freqtrade.freqtradebot - INFO - Cancelling current stoploss on exchange for pair APE/USDT:USDT (orderid:39152692) in order to add another one ...
```

diff of one of the positions (see `|` for the changed lines):
```
  {                                               {
    "info": {                                       "info": {
      "value": "342.5344",                            "value": "342.5344",
      "leverage": "2",                                "leverage": "2",
      "mode": "single",                               "mode": "single",
      "realised_point": "0",                          "realised_point": "0",
      "contract": "FTM_USDT",                         "contract": "FTM_USDT",
      "entry_price": "0.4833",                        "entry_price": "0.4833",
      "mark_price": "0.4784",                         "mark_price": "0.4784",
      "history_point": "0",                           "history_point": "0",
      "realised_pnl": "-0.1730214",                   "realised_pnl": "-0.1730214",
      "close_order": null,                            "close_order": null,
      "size": "716",                                  "size": "716",
      "cross_leverage_limit": "0",                    "cross_leverage_limit": "0",
      "pending_orders": "0",                          "pending_orders": "0",
      "adl_ranking": "5",                             "adl_ranking": "5",
      "maintenance_rate": "0.025",                    "maintenance_rate": "0.025",
      "unrealised_pnl": "-3.5084",                    "unrealised_pnl": "-3.5084",
      "user": "11853388",                             "user": "11853388",
      "leverage_max": "20",                           "leverage_max": "20",
      "history_pnl": "-26.015948589534",              "history_pnl": "-26.015948589534",
      "risk_limit": "500000",                         "risk_limit": "500000",
      "margin": "175.3899638",                |       "margin": "175.0325008",
      "last_close_pnl": "-2.300428934506",            "last_close_pnl": "-2.300428934506",
      "liq_price": "0.2447"                   |       "liq_price": "0.2452"
    },                                              },
    "id": null,                                     "id": null,
    "symbol": "FTM/USDT:USDT",                      "symbol": "FTM/USDT:USDT",
    "timestamp": null,                              "timestamp": null,
    "datetime": null,                               "datetime": null,
    "initialMargin": 171.5241008,                   "initialMargin": 171.5241008,
    "initialMarginPercentage": 0.50075,             "initialMarginPercentage": 0.50075,
    "maintenanceMargin": 8.56336,                   "maintenanceMargin": 8.56336,
    "maintenanceMarginPercentage": 0.025,           "maintenanceMarginPercentage": 0.025,
    "entryPrice": 0.4833,                           "entryPrice": 0.4833,
    "notional": 342.5344,                           "notional": 342.5344,
    "leverage": 2.0,                                "leverage": 2.0,
    "unrealizedPnl": -3.5084,                       "unrealizedPnl": -3.5084,
    "contracts": 716.0,                             "contracts": 716.0,
    "contractSize": 1.0,                            "contractSize": 1.0,
    "marginRatio": null,                            "marginRatio": null,
    "liquidationPrice": 0.2447,               |     "liquidationPrice": 0.2452,
    "markPrice": 0.4784,                            "markPrice": 0.4784,
    "collateral": 175.3899638,                |     "collateral": 175.0325008,
    "marginMode": "isolated",                       "marginMode": "isolated",
    "side": "long",                                 "side": "long",
    "percentage": -2.045426843013072                "percentage": -2.045426843013072
  },                                              },

```

Diff of the stop order:
```
  {                                               {
    "id": "39152688",                         |     "id": "39153069",
    "clientOrderId": null,                          "clientOrderId": null,
    "timestamp": 1679433013000,               |     "timestamp": 1679433611000,
    "datetime": "2023-03-21T21:10:13.000Z",   |     "datetime": "2023-03-21T21:20:11.000Z",
    "lastTradeTimestamp": 1679433013000,      |     "lastTradeTimestamp": 1679433611000,
    "status": "open",                               "status": "open",
    "symbol": "FTM/USDT:USDT",                      "symbol": "FTM/USDT:USDT",
    "type": "limit",                                "type": "limit",
    "timeInForce": "GTC",                           "timeInForce": "GTC",
    "postOnly": false,                              "postOnly": false,
    "reduceOnly": null,                             "reduceOnly": null,
    "side": "sell",                                 "side": "sell",
    "price": 0.4606,                          |     "price": 0.4608,
    "stopPrice": 0.4652,                      |     "stopPrice": 0.4654,
    "triggerPrice": 0.4652,                   |     "triggerPrice": 0.4654,
    "average": null,                                "average": null,
    "amount": 716.0,                                "amount": 716.0,
    "cost": 0.0,                                    "cost": 0.0,
    "filled": 0.0,                                  "filled": 0.0,
    "remaining": 716.0,                             "remaining": 716.0,
    "fee": null,                                    "fee": null,
    "fees": [],                                     "fees": [],
    "trades": [],                                   "trades": [],
    "info": {                                       "info": {
      "user": "11853388",                             "user": "11853388",
      "trigger": {                                    "trigger": {
        "strategy_type": "0",                           "strategy_type": "0",
        "price_type": "0",                              "price_type": "0",
        "price": "0.4652",                    |         "price": "0.4654",
        "rule": "2",                                    "rule": "2",
        "expiration": "0"                               "expiration": "0"
      },                                              },
      "initial": {                                    "initial": {
        "contract": "FTM_USDT",                         "contract": "FTM_USDT",
        "size": "-716",                                 "size": "-716",
        "price": "0.4606",                    |         "price": "0.4608",
        "tif": "gtc",                                   "tif": "gtc",
        "text": "",                                     "text": "",
        "iceberg": "0",                                 "iceberg": "0",
        "is_close": false,                              "is_close": false,
        "is_reduce_only": true,                         "is_reduce_only": true,
        "auto_size": ""                                 "auto_size": ""
      },                                              },
      "id": "39152688",                       |       "id": "39153069",
      "trade_id": "0",                                "trade_id": "0",
      "status": "open",                               "status": "open",
      "reason": "",                                   "reason": "",
      "create_time": "1679433013",            |       "create_time": "1679433611",
      "finish_time": "1679433013",            |       "finish_time": "1679433611",
      "is_stop_order": false,                         "is_stop_order": false,
      "stop_trigger": {                               "stop_trigger": {
        "rule": "0",                                    "rule": "0",
        "trigger_price": "",                            "trigger_price": "",
        "order_price": ""                               "order_price": ""
      },                                              },
      "me_order_id": "0",                             "me_order_id": "0",
      "order_type": ""                                "order_type": ""
    }                                               }
  },                                              },
```

Full balance diff:
```
{                                               {
  "info": [                                       "info": [
    {                                               {
      "order_margin": "0",                            "order_margin": "0",
      "point": "0",                                   "point": "0",
      "bonus": "0",                                   "bonus": "0",
      "history": {                                    "history": {
        "dnw": "1081.2956",                             "dnw": "1081.2956",
        "pnl": "-138.04488973154",                      "pnl": "-138.04488973154",
        "refr": "0",                                    "refr": "0",
        "point_fee": "0",                               "point_fee": "0",
        "point_dnw": "0",                               "point_dnw": "0",
        "bonus_dnw": "0",                               "bonus_dnw": "0",
        "point_refr": "0",                              "point_refr": "0",
        "bonus_offset": "0",                            "bonus_offset": "0",
        "cross_settle": "0",                            "cross_settle": "0",
        "fee": "-69.5545510365",                        "fee": "-69.5545510365",
        "fund": "0.731873593676"                        "fund": "0.731873593676"
      },                                              },
      "unrealised_pnl": "9.32046",            |       "unrealised_pnl": "8.98076",
      "total": "874.428032825636",                    "total": "874.428032825636",
      "available": "6.671774538136",          |       "available": "7.029237538136",
      "enable_credit": false,                         "enable_credit": false,
      "in_dual_mode": false,                          "in_dual_mode": false,
      "currency": "USDT",                             "currency": "USDT",
      "position_margin": "867.7562582875",    |       "position_margin": "867.3987952875",
      "user": "11853388"                              "user": "11853388"
    }                                               }
  ],                                              ],
  "USDT": {                                       "USDT": {
    "free": 6.671774538136,                   |     "free": 7.029237538136,
    "used": 867.7562582875,                   |     "used": 867.3987952875,
    "total": 874.428032825636                       "total": 874.428032825636
  },                                              },
  "free": {                                       "free": {
    "USDT": 6.671774538136                    |     "USDT": 7.029237538136
  },                                              },
  "used": {                                       "used": {
    "USDT": 867.7562582875                    |     "USDT": 867.3987952875
  },                                              },
  "total": {                                      "total": {
    "USDT": 874.428032825636                        "USDT": 874.428032825636
  }                                               }
}                                               }

```

Later, `free` increased a little more, and then decreased. 

```
2023-03-21_22.30.15 balance USDT:      {"free": 7.172222738136, "used": 867.2558100875, "total": 874.428032825636}
2023-03-21_22.30.21 balance USDT:      {"free": 6.564535638136, "used": 867.8634971875, "total": 874.428032825636}
```
Corresponding freqtrade log (timestams are 1 hour behind):

```
2023-03-21 21:30:12,747 - freqtrade.freqtradebot - INFO - Cancelling current stoploss on exchange for pair SOL/USDT:USDT (orderid:39153239) in order to add another one ...
2023-03-21 21:30:14,274 - freqtrade.freqtradebot - INFO - Cancelling current stoploss on exchange for pair BTC/USDT:USDT (orderid:39153240) in order to add another one ...
2023-03-21 21:30:15,824 - freqtrade.freqtradebot - INFO - Cancelling current stoploss on exchange for pair FTM/USDT:USDT (orderid:39153242) in order to add another one ...
2023-03-21 21:30:17,339 - freqtrade.freqtradebot - INFO - Cancelling current stoploss on exchange for pair DOT/USDT:USDT (orderid:39153243) in order to add another one ...
2023-03-21 21:30:18,860 - freqtrade.freqtradebot - INFO - Cancelling current stoploss on exchange for pair APE/USDT:USDT (orderid:39153245) in order to add another one ...
```

diff of one of the positions, (see `|` for the changed lines):

```
  {                                               {
    "info": {                                       "info": {
      "value": "341.3888",                    |       "value": "341.6036",
      "leverage": "2",                                "leverage": "2",
      "mode": "single",                               "mode": "single",
      "realised_point": "0",                          "realised_point": "0",
      "contract": "FTM_USDT",                         "contract": "FTM_USDT",
      "entry_price": "0.4833",                        "entry_price": "0.4833",
      "mark_price": "0.4768",                 |       "mark_price": "0.4771",
      "history_point": "0",                           "history_point": "0",
      "realised_pnl": "-0.1730214",                   "realised_pnl": "-0.1730214",
      "close_order": null,                            "close_order": null,
      "size": "716",                                  "size": "716",
      "cross_leverage_limit": "0",                    "cross_leverage_limit": "0",
      "pending_orders": "0",                          "pending_orders": "0",
      "adl_ranking": "5",                             "adl_ranking": "5",
      "maintenance_rate": "0.025",                    "maintenance_rate": "0.025",
      "unrealised_pnl": "-4.654",             |       "unrealised_pnl": "-4.4392",
      "user": "11853388",                             "user": "11853388",
      "leverage_max": "20",                           "leverage_max": "20",
      "history_pnl": "-26.015948589534",              "history_pnl": "-26.015948589534",
      "risk_limit": "500000",                         "risk_limit": "500000",
      "margin": "174.8895156",                |       "margin": "175.4972027",
      "last_close_pnl": "-2.300428934506",            "last_close_pnl": "-2.300428934506",
      "liq_price": "0.2454"                   |       "liq_price": "0.2445"
    },                                              },
    "id": null,                                     "id": null,
    "symbol": "FTM/USDT:USDT",                      "symbol": "FTM/USDT:USDT",
    "timestamp": null,                              "timestamp": null,
    "datetime": null,                               "datetime": null,
    "initialMargin": 170.9504416,             |     "initialMargin": 171.0580027,
    "initialMarginPercentage": 0.50075,             "initialMarginPercentage": 0.50075,
    "maintenanceMargin": 8.53472,             |     "maintenanceMargin": 8.54009,
    "maintenanceMarginPercentage": 0.025,           "maintenanceMarginPercentage": 0.025,
    "entryPrice": 0.4833,                           "entryPrice": 0.4833,
    "notional": 341.3888,                     |     "notional": 341.6036,
    "leverage": 2.0,                                "leverage": 2.0,
    "unrealizedPnl": -4.654,                  |     "unrealizedPnl": -4.4392,
    "contracts": 716.0,                             "contracts": 716.0,
    "contractSize": 1.0,                            "contractSize": 1.0,
    "marginRatio": null,                            "marginRatio": null,
    "liquidationPrice": 0.2454,               |     "liquidationPrice": 0.2445,
    "markPrice": 0.4768,                      |     "markPrice": 0.4771,
    "collateral": 174.8895156,                |     "collateral": 175.4972027,
    "marginMode": "isolated",                       "marginMode": "isolated",
    "side": "long",                                 "side": "long",
    "percentage": -2.7224264274728847         |     "percentage": -2.595143126852375
  },                                              },

```

Diff of the stop order:
```
  {                                               {
    "id": "39153242",                         |     "id": "39153399",
    "clientOrderId": null,                          "clientOrderId": null,
    "timestamp": 1679433910000,               |     "timestamp": 1679434216000,
    "datetime": "2023-03-21T21:25:10.000Z",   |     "datetime": "2023-03-21T21:30:16.000Z",
    "lastTradeTimestamp": 1679433910000,      |     "lastTradeTimestamp": 1679434216000,
    "status": "open",                               "status": "open",
    "symbol": "FTM/USDT:USDT",                      "symbol": "FTM/USDT:USDT",
    "type": "limit",                                "type": "limit",
    "timeInForce": "GTC",                           "timeInForce": "GTC",
    "postOnly": false,                              "postOnly": false,
    "reduceOnly": null,                             "reduceOnly": null,
    "side": "sell",                                 "side": "sell",
    "price": 0.461,                           |     "price": 0.4612,
    "stopPrice": 0.4656,                      |     "stopPrice": 0.4658,
    "triggerPrice": 0.4656,                   |     "triggerPrice": 0.4658,
    "average": null,                                "average": null,
    "amount": 716.0,                                "amount": 716.0,
    "cost": 0.0,                                    "cost": 0.0,
    "filled": 0.0,                                  "filled": 0.0,
    "remaining": 716.0,                             "remaining": 716.0,
    "fee": null,                                    "fee": null,
    "fees": [],                                     "fees": [],
    "trades": [],                                   "trades": [],
    "info": {                                       "info": {
      "user": "11853388",                             "user": "11853388",
      "trigger": {                                    "trigger": {
        "strategy_type": "0",                           "strategy_type": "0",
        "price_type": "0",                              "price_type": "0",
        "price": "0.4656",                    |         "price": "0.4658",
        "rule": "2",                                    "rule": "2",
        "expiration": "0"                               "expiration": "0"
      },                                              },
      "initial": {                                    "initial": {
        "contract": "FTM_USDT",                         "contract": "FTM_USDT",
        "size": "-716",                                 "size": "-716",
        "price": "0.461",                     |         "price": "0.4612",
        "tif": "gtc",                                   "tif": "gtc",
        "text": "",                                     "text": "",
        "iceberg": "0",                                 "iceberg": "0",
        "is_close": false,                              "is_close": false,
        "is_reduce_only": true,                         "is_reduce_only": true,
        "auto_size": ""                                 "auto_size": ""
      },                                              },
      "id": "39153242",                       |       "id": "39153399",
      "trade_id": "0",                                "trade_id": "0",
      "status": "open",                               "status": "open",
      "reason": "",                                   "reason": "",
      "create_time": "1679433910",            |       "create_time": "1679434216",
      "finish_time": "1679433910",            |       "finish_time": "1679434216",
      "is_stop_order": false,                         "is_stop_order": false,
      "stop_trigger": {                               "stop_trigger": {
        "rule": "0",                                    "rule": "0",
        "trigger_price": "",                            "trigger_price": "",
        "order_price": ""                               "order_price": ""
      },                                              },
      "me_order_id": "0",                             "me_order_id": "0",
      "order_type": ""                                "order_type": ""
    }                                               }
  },                                              },

```

Full balance diff:
```
{                                               {
  "info": [                                       "info": [
    {                                               {
      "order_margin": "0",                            "order_margin": "0",
      "point": "0",                                   "point": "0",
      "bonus": "0",                                   "bonus": "0",
      "history": {                                    "history": {
        "dnw": "1081.2956",                             "dnw": "1081.2956",
        "pnl": "-138.04488973154",                      "pnl": "-138.04488973154",
        "refr": "0",                                    "refr": "0",
        "point_fee": "0",                               "point_fee": "0",
        "point_dnw": "0",                               "point_dnw": "0",
        "bonus_dnw": "0",                               "bonus_dnw": "0",
        "point_refr": "0",                              "point_refr": "0",
        "bonus_offset": "0",                            "bonus_offset": "0",
        "cross_settle": "0",                            "cross_settle": "0",
        "fee": "-69.5545510365",                        "fee": "-69.5545510365",
        "fund": "0.731873593676"                        "fund": "0.731873593676"
      },                                              },
      "unrealised_pnl": "9.30272",            |       "unrealised_pnl": "9.57706",
      "total": "874.428032825636",                    "total": "874.428032825636",
      "available": "7.172222738136",          |       "available": "6.564535638136",
      "enable_credit": false,                         "enable_credit": false,
      "in_dual_mode": false,                          "in_dual_mode": false,
      "currency": "USDT",                             "currency": "USDT",
      "position_margin": "867.2558100875",    |       "position_margin": "867.8634971875",
      "user": "11853388"                              "user": "11853388"
    }                                               }
  ],                                              ],
  "USDT": {                                       "USDT": {
    "free": 7.172222738136,                   |     "free": 6.564535638136,
    "used": 867.2558100875,                   |     "used": 867.8634971875,
    "total": 874.428032825636                       "total": 874.428032825636
  },                                              },
  "free": {                                       "free": {
    "USDT": 7.172222738136                    |     "USDT": 6.564535638136
  },                                              },
  "used": {                                       "used": {
    "USDT": 867.2558100875                    |     "USDT": 867.8634971875
  },                                              },
  "total": {                                      "total": {
    "USDT": 874.428032825636                        "USDT": 874.428032825636
  }                                               }
}                                               }
```

GateIO browser messages (had a screen recorder running):
`Sell xxx FTM FTM/USDT contracts at xxxUSDT. Trigger price: Last Price < xxxx. This Stop Order is Canclled` (sic)
`Sell xxx FTM FTM/USDT contracts at xxxUSDT. Trigger price: Last Price < xxxx. This Stop Order is Created`

So when deleting / creating stop orders, balance gets moved between `used` and `free`, but `total` seems to be staying the same. 

Any thoughts.. ?

**Comment by gaardiolor** [comment]
Also worth noting, I've been running this strategy for months without these issues.. Did some more investigation, this started shortly after adding balance and changing leverage from 1 to 2.

**Comment by xmatthias** [comment]
no idea
i'm pretty certain that values will behave exactly the same when you're trading manually on the exchange (obviously using the same actions / configurations) - so to me, this seems to be an oddity / bug in the gate platform - one i can't for now explain based on the help alone.
Obviously it won't happen without leverage - as then as then you're not borrowing anything from the exchange.

If you look at the [isolated margin mode](https://www.gate.io/help/futures/futures_logic/22080/isolated-margin-cross-margin) doc page, you'll find no mention of changes to the position's collateral.

balance total will always stay identical in futures mode (unless you close it) - as you technically still have the USDT collateral in your account (as used) - it's just no longer "free" to use. Total will only change once you actually close a position (with a non-breakeven result).

---

the changes to the stop order are as expected for a trailing stop i'd say.
Now what's surprising me a bit is the fact that the position value (this explains the fluctuation of the balance) changes.

this seems wrong to me (no order was actually filled as far as i can see) - and something you should probably clarify with gate support. 
It looks like a bug in the gate platform to me ... and i'd not know how freqtrade can (or should) influence this behavior.
